internal/infrastructure/repository/contest_impl.go

```diff
@@ -120,7 +131,7 @@ func (r *ContestRepository) UpdateContest(ctx context.Context, contestID uuid.UU
 	if v, ok := args.Since.V(); ok {
 		changes["since"] = v
 	}
-	if v, ok := args.Until.V(); ok {
+	if v, ok := args.Until.V(); ok == untilEmpty || v != origin.Until {
```

Comment:
`time.Time`の比較なので`.Equal`ではないでしょうか？

```suggestion
	if v, ok := args.Until.V(); ok == untilEmpty || v.Equal(origin.Until) {
```